app/cp_nigeria/views.py

```diff
@@ -140,7 +140,51 @@ def cpn_demand_params(request, proj_id, step_id=STEP_MAPPING["demand_profile"]):
     # TODO change DB default value to 1
     # TODO include the possibility to display the "expected_consumer_increase", "expected_demand_increase" fields
     # with option advanced_view set by user choice
-    form = ConsumerGroupForm(initial={"number_consumers": 1}, advanced_view=False)
+    # TODO@Paula handle form deletion
+    # TODO when page is reloaded, empty extra form is generated (can be deleted but is annoying)
+    if request.method == "POST":
+        if project is not None:
+            formset_qs = ConsumerGroup.objects.filter(project=project)
+            formset = ConsumerGroupFormSet(request.POST, queryset=formset_qs)
+
+        else:
+            formset = ConsumerGroupFormSet(request.POST)
+
+        for form in formset:
+            import pdb; pdb.set_trace()
+            prefix = form.prefix
+            group_id = int(form.prefix[-1:])
+            # set timeseries queryset so form doesn't throw a validation error
```

Comment:
You can also set the field with `blank=True` in the model, that way the form does not show a validation error

Reply:
this adjustment seems to be necessary in this case because of the dependent dropdown in the demand input (timeseries dynamically filtered depending on the consumer type)